src/ast/postprocessing/structured_documentation_reconstruction.ts

```diff
@@ -146,19 +196,47 @@ export class StructuredDocumentationReconstructingPostprocessor
             return;
         }
 
-        const structDocNode = this.reconstructor.process(node, source);
+        /**
+         * Skip reconstructing preceding strcutured documentation
+         * when related fields is already an instance of StructuredDocumentation.
+         */
+        if (!(node.documentation instanceof StructuredDocumentation)) {
+            const precedingGap = this.reconstructor.getPrecedingGapCoordinates(node);
+            const preceding = this.reconstructor.fragmentCoordsToStructDoc(precedingGap, source);
 
-        if (structDocNode === undefined) {
-            return;
+            if (preceding) {
+                preceding.id = context.lastId + 1;
+
+                context.register(preceding);
+
+                node.documentation = preceding;
+
+                if (preceding.parent !== node) {
```

Comment:
Is this `if` neccessary? Seems like you can always just do `preceding.parent = node;`

Reply:
Will address

Reply:
Addressed in https://github.com/ConsenSys/solc-typed-ast/pull/115/commits/872be86fefb3d356d2552929b0a260684ffd6801